tap_redshift/__init__.py

```diff
@@ -110,24 +111,28 @@ def discover_catalog(conn, db_schema):
             db_name, db_schema, cols, is_view, key_properties)
         tap_stream_id = '{}.{}'.format(
             db_name, qualified_table_name)
-        entry = CatalogEntry(
-            tap_stream_id=tap_stream_id,
-            stream=table_name,
-            schema=schema,
-            table=qualified_table_name,
-            metadata=metadata)
+        entry = {
+            'tap_stream_id': tap_stream_id,
+            'table_name': qualified_table_name,
+            'schema': schema.to_dict(),
+            'stream': table_name,
+            'metadata': metadata,
+            'column_order': [str(column) for column in schema.properties]
```

Comment:
@jayma91, at this point, the column order can be derived from schema.properties, and then is stuffed in a new `column_order` parameter. But won't we have access to schema.properties later on?

Instead of storing the new `column_order` parameter, why not just pull it straight from schema.properties wherever you need the column order?

Reply:
@mvanlamz I was informed that there was a previous issue with connectors in general where the row data didn't match the column order, which was happening in our backend code, not in the tap code. The fix for the previous issue was to use column_order property and use Object.keys(schema.schema.properties) only if column_order is undefined in our backend code. I suspect that Object.keys was distorting the order of columns in our backend by deriving different order than the order in catalog discovered by tap, causing mismatch for columns and row data. For this reason, I thought it would be safer to follow using column_order.